src/bundles/communication/Communications.ts

```diff
@@ -0,0 +1,123 @@
+import context from 'js-slang/context';
+import { MultiUserController } from './MultiUserController';
+import { GlobalStateController } from './GlobalStateController';
+import { RpcController } from './RpcController';
+
+class CommunicationModuleState {
+  multiUser: MultiUserController;
+  globalState: GlobalStateController | null = null;
+  rpc: RpcController | null = null;
+
+  constructor(address: string, port: number) {
+    let multiUser = new MultiUserController();
+    multiUser.setupController(address, port);
+    this.multiUser = multiUser;
+  }
+}
+
+let moduleState = new CommunicationModuleState('broker.hivemq.com', 8884);
+context.moduleContexts.communication.state = moduleState;
+
+// Loop
+
+let interval: number | undefined;
+
+/**
+ * Keeps the program running so that messages can come in.
+ */
+export function keepRunning() {
+  interval = window.setInterval(() => {}, 20000);
+}
+
+/**
+ * Removes interval that keeps the program running.
+ */
+export function stopRunning() {
+  if (interval) {
```

Comment:
Is the ID returned from `setInterval` always greater than 0? If not (or unsure), I think it would be best to explicitly state `interval !== undefined` as the condition.

Reply:
What @RichDom2185 means is that if `interval` can become 0 (also considered `false`) you should write `interval !== undefined`.

Reply:
Ok fixed